src/Database/Query/SelectQuery.php

```diff
@@ -89,4 +90,18 @@ public function set($key, $value = null, $types = [])
 
         return parent::set($key, $value, $types);
     }
+
+    /**
+     * Sets the connection role.
+     *
+     * @param string $role Connection role ('read' or 'write')
+     * @return $this
+     */
+    public function setConnectionRole(string $role)
```

Comment:
We could move `useReadRole()` and `useWriteRole()` here.

Reply:
We can. I was trying to keep "connection" in the name to make it clear. I thought `useReadConnectionRole()` was too much, but no reason we can't add it.

Reply:
What about `useReadConnection()`?

Reply:
We could but since this is now self-contained within a single connection, I thought that would be misleading.

Reply:
I'll add whatever you think is best here. This is just the though process I went through.

Reply:
Lets go with `useReadRole()` and `useWriteRole()` then. It saves userland code having to import Connection or ConnectionInterface to get the constants.